main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -129,6 +129,7 @@ object Resolver extends Phase[NamedAst.Root, ResolvedAst.Root] {
         for {
           tparams <- tparamsVal
           t <- lookupType(tpe, ns0, prog0)
+          _ <- checkProperType(t)
```

Comment:
I think it is interesting to note that the types of a case should not be uninhibited, but the type parameters of an enum could be. For example, when you have phantom type parameters.